lib/sdk11/components/libraries/bootloader_dfu/bootloader.c

```diff
@@ -336,7 +337,9 @@ uint32_t bootloader_dfu_start(bool ota, uint32_t timeout_ms, bool cancel_timeout
       err_code = dfu_transport_ble_update_start();
     }else
     {
-      // timeout_ms > 0 is forced startup DFU
+      // DFU mode with timeout can be
+      // - Forced startup DFU for nRF52832 or
+      // - Makecode single tap reset but no enumerated (battery power)
       if ( timeout_ms )
       {

```

Comment:
It looks like you removed the timeout time-keeping code, but why is there still a `timeout_ms` mention here, and mentions of timeout in the comments? I am just a little confused.

Reply:
This is another timeout, which is different than the overall timeout which is just removed. This one is used by nrf52832, 832 always go into DFU mode for 1 second when power on. This is required for Arduino auto reset is hardware to CP2104 DTR. It is later used by Makecode to exit its singletap reset to DFU when on battery power only (no usb enumeration) after 3 seconds.

Reply:
:heavy_check_mark: